### DISCUSSION TITLE
audioread>readaudio errors preventing auto review + classification functions

### DISCUSSION THREAD

**Discussion by PenguinPeeps** [DrCoffey/DeepSqueak#219] [Q&A]
I'm hoping I can get some thoughts on this error thread below. I get this same error when I try to run an unsupervised and unsupervised classification, as well as when I try to batch reject by a score threshold. I've tried moving around some files to see if I can identify the issue, but the error is persistent regardless.

Error using audioread>readaudio
The filename specified was not found in the MATLAB path.

Error in audioread (line 160)
    [y, Fs] = readaudio (filename, range, datatype);

Error in squeakData/AudioSamples (line 100)
                obj.StoredSamples = audioread(obj.audiodata.Filename, [obj.AudioStartSample, obj.AudioStopSample]);

Error in CreateFocusSpectrogram (line 35)
    audio = audioReader.AudioSamples(box(1), box(1) + box(3));

Error in Batch_Reject_by_Threshold_Callback (line 96)
        [I,wind,noverlap,nfft,rate,box,~,~,~,~,pow] = CreateFocusSpectrogram(Calls(i,:), handles, true, [], audioReader);

Error in gui_mainfcn (line 95)
        feval(varargin{:});

Error in DeepSqueak (line 29)
    gui_mainfcn(gui_State, varargin{:});

Error in matlab.graphics.internal.figfile.FigFile/read>@(hObject,eventdata)DeepSqueak('Batch_Reject_by_Threshold_Callback',hObject,eventdata,guidata(hObject))
 
Error while evaluating Menu Callback.

Error using audioread>readaudio
The filename specified was not found in the MATLAB path.

Error in audioread (line 160)
    [y, Fs] = readaudio (filename, range, datatype);

Error in squeakData/AudioSamples (line 100)
                obj.StoredSamples = audioread(obj.audiodata.Filename, [obj.AudioStartSample, obj.AudioStopSample]);

Error in CreateFocusSpectrogram (line 35)
    audio = audioReader.AudioSamples(box(1), box(1) + box(3));

Error in Batch_Reject_by_Threshold_Callback (line 96)
        [I,wind,noverlap,nfft,rate,box,~,~,~,~,pow] = CreateFocusSpectrogram(Calls(i,:), handles, true, [], audioReader);

Error in gui_mainfcn (line 95)
        feval(varargin{:});

Error in DeepSqueak (line 29)
    gui_mainfcn(gui_State, varargin{:});

Error in matlab.graphics.internal.figfile.FigFile/read>@(hObject,eventdata)DeepSqueak('Batch_Reject_by_Threshold_Callback',hObject,eventdata,guidata(hObject))
 
Error while evaluating Menu Callback.

**Comment by DrCoffey** [comment]
Hey @PenguinPeeps, did you select the correct audio folder before trying to run these tools? It looks like audioread can't find the file path.

**Reply by PenguinPeeps** [reply]
Hello!

I was only working with one audio folder at the time. Unfortunately, I couldn't determine the source of the error, so I created a new audio folder.

**Comment by bfsinger** [comment]
Hi all,

I am experiencing what seems to be the same issue described here. I am having a lot of difficulty getting unsupervised clustering to work.

The original analysis was done on a different computer. My detection and audio files are now stored on an external hard drive. Despite using the built-in tools to adjust the audio and detection folders, I receive the 'filename not found' error below when attempting to run the clustering. It seems the software is failing to correctly reference the audio path on the external volume during the spectrogram creation phase. I've tried this on both a mac and pc.

Any advice or workarounds would be greatly appreciated.

___

Error using audioread>readaudio (line 160)
The filename specified was not found in the MATLAB path.

Error in audioread (line 153)
    [y, Fs] = readaudio (filename, fileNameObj.RemoteFileName, range, datatype);
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
Error in squeakData/AudioSamples (line 100)
                obj.StoredSamples = audioread(obj.audiodata.Filename, [obj.AudioStartSample, obj.AudioStopSample]);
                                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
Error in CreateFocusSpectrogram (line 35)
    audio = audioReader.AudioSamples(box(1), box(1) + box(3));
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
Error in CreateClusteringData (line 115)
    [I,wind,noverlap,nfft,rate,box,s,fr,ti,~,pow] = CreateFocusSpectrogram(Calls(i,:), handles, true, [], audioReader);
                                                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
Error in create_VAE_model (line 6)
[ClusteringData, clustAssign, options.freqRange, options.maxDuration, options.spectrogram] = CreateClusteringData(handles, 'scale_duration', true, 'fixed_frequency', true,'forClustering', true, 'save_data', true);
                                                                                             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
Error in UnsupervisedClustering_Callback (line 34)
                            [encoderNet, decoderNet, options, ClusteringData] = create_VAE_model(handles);
                                                                                ^^^^^^^^^^^^^^^^^^^^^^^^^
Error in gui_mainfcn (line 95)
        feval(varargin{:});
        ^^^^^^^^^^^^^^^^^^
Error in DeepSqueak (line 29)
    gui_mainfcn(gui_State, varargin{:});
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
Error in matlab.graphics.internal.figfile.FigFile/read>@(hObject,eventdata)DeepSqueak('UnsupervisedClustering_Callback',hObject,eventdata,guidata(hObject))
 

Error using matlab.ui.internal.controller.WebMenuController/fireMenuSelectedEvent (line 85)
Error while evaluating Menu Callback.
